cmake/EnableSanitizers.cmake

```diff
@@ -36,46 +36,91 @@ endfunction()
 
 option(SANITIZE_ADDRESS "Enable AddressSanitizer for the build." "")
 if (SANITIZE_ADDRESS)
+    # Try each group of options and pick the first one that works.
     set(ASAN_CANDIDATES "-fsanitize=address -fno-omit-frame-pointer" "-fsanitize=address")
     sanitizer_test(AddressSanitizer ${ASAN_CANDIDATES})
+    if (NOT AddressSanitizer_ENABLED)
+        message(FATAL_ERROR "AddressSanitizer requested but could not be enabled."
```

Comment:
If these messages will be concatenated as-is (I presume string literals are), you need to add a space at the end of each of these lines (except the last one) so that the resulting output is readable. You did this in the other sanitizer message blocks.

Reply:
Doh! Fixed.